### PR TITLE
[Python] Add Flight streaming example

### LINKED ISSUE #86: [Python][Flight] Recipe to show how to stream data in `do_get`
The current flight recipe loads in memory the whole content of the datasets before writing them to the response, it would be good to have a recipe that also shows how to stream data for the case of big datasets that might not fit in memory.

### PR DESCRIPTION
**Pull request by lidavidm**
Fixes #86.

### CODE REVIEW

**Comment by lidavidm** [comment]
<details>
<summary>Rendered</summary>

image

</details>

**Comment by lidavidm** [comment]
CC @amol- if you have any comments here.

File: python/source/flight.rst [hunk 1]
```diff
@@ -226,3 +226,198 @@ we might list all parquet files that are currently stored by the server:
...
+           with dataset_path.open("wb") as sink:
+               with pa.parquet.ParquetWriter(sink, reader.schema) as writer:
+                   for chunk in reader:
+                       writer.write_table(pa.Table.from_batches([chunk.data]))
+
+       def do_get(self, context, ticket):
+           dataset = ticket.ticket.decode('utf-8')
+
+           # Stream data from a generator we implement
+           if dataset == ":random:":
```
Comment (amol-) [inline python/source/flight.rst:289]: iI think that the `:random:` special dataset might be confusing for a reader in the context of this recipe. It's great for playing around, but the recipe goal was to read back datasets you stored, So a reader might wonder what's the purpose of that code block and get confused by it.

Comment (lidavidm) [reply python/source/flight.rst:289]: Makes sense. I wanted to demonstrate GeneratorStream, so instead I'll use it to replace RecordBatchStream and log a message or something.

Comment (amol-) [reply python/source/flight.rst:289]: I think that might be its own recipe, like "How to stream data generated on the fly". At that point it  could also be a smaller example with just `do_get`

Comment (lidavidm) [reply python/source/flight.rst]: That makes sense. I've removed that part of the example.

**Comment by amol-** [comment]
What's the state of this recipe? Does it need further review or work?

**Comment by lidavidm** [comment]
Sorry for the delay - I did update it based on your feedback, so I think it's ready for another look.

**Pull request merged**